corehq/apps/smsbillables/tests/test_gateway_fees.py

```diff
@@ -233,6 +233,24 @@ def test_no_matching_fee(self):
                 self.assertIsNotNone(billable)
                 self.assertIsNone(billable.gateway_fee)
 
+    def test_fee_for_non_global_backend(self):
+        self.create_least_specific_gateway_fees()
+
+        private_backend_ids = generator.arbitrary_non_global_backend_ids()
+        messages = generator.arbitrary_messages_by_backend_and_direction(
+            private_backend_ids)
+        arbitrary_message = messages[randint(0, len(messages))]
+
+        toggles.ENABLE_INCLUDE_SMS_GATEWAY_CHARGING.set(
+            arbitrary_message.domain, True, toggles.NAMESPACE_DOMAIN)
+        billable = SmsBillable.create(arbitrary_message)
+        toggles.ENABLE_INCLUDE_SMS_GATEWAY_CHARGING.set(
+            arbitrary_message.domain, False, toggles.NAMESPACE_DOMAIN)
+
+        self.assertIsNotNone(billable)
+        self.assertIsNotNone(billable.gateway_fee)
```

Comment:
I would also be good to check that the gateway fee charge assigned is correct.

Here's an example of how to mock a response from the twilio api:

https://github.com/dimagi/commcare-hq/blob/1bfe88ffe51eee2424a48e998b5f09926096f2c3/corehq/apps/smsbillables/tests/test_gateway_fees.py#L236-L272